### PR TITLE
fix: Add ContractAddress type to consolidate contract id parsing with aliases

### PR DESCRIPTION
**Pull request by willemneal**
New type is similar to Address (which later should be combined since contract addresses can be used with `contract invoke` contract method args.

### CODE REVIEW

**Review by leighmcculloch — APPROVED** [review]

**Comment by willemneal** [comment]
It seems that this is not needed since it is resolved in the function that fetches the wasm. However, given the work that @fnando is doing with the config it might be worth adding a similar `ContractAddress` type.

**Comment by Ifropc** [comment]
Change seems to be super simple but looks like there are some test failures cause of error types

File: cmd/soroban-cli/src/config/alias.rs [hunk 1]
```diff
@@ -1,8 +1,49 @@
...
 }
+
+/// Address can be either a contract address, C.. or eventually an alias of a contract address.
+#[derive(Clone, Debug)]
+pub enum ContractAddress {
+    ContractId(stellar_strkey::Contract),
+    Alias(String),
+}
+
+impl Default for ContractAddress {
```
Comment (Ifropc) [inline cmd/soroban-cli/src/config/alias.rs:19]: Does ContractAddress need to have a default? 🤔

**Review by Ifropc — COMMENTED** [review]
Love the addition of the new enum, one small question and LGTM

File: cmd/soroban-cli/src/config/alias.rs (hunk 1 above)
Comment (willemneal) [reply cmd/soroban-cli/src/config/alias.rs:19]: Good question I copy/pasted from address::Address. I'll look into removing it.

**Comment by leighmcculloch** [comment]
Converted to draft because builds seem to be broken.

File: cmd/soroban-cli/src/config/alias.rs (hunk 1 above)
Comment (willemneal) [reply cmd/soroban-cli/src/config/alias.rs:19]: Checked and some of the commands that use it require it.

**Comment by willemneal** [comment]
@Ifropc

**Review by Ifropc — APPROVED** [review]
LGTM!

**Pull request merged**